docs/source/integrations/deltalake_versioning.md

```diff
@@ -0,0 +1,214 @@
+
+```python
+In [1]: model_input_table = catalog.datasets['model_input_table']
+```
+You can inspect the history of the Delta table by accessing the `history` attribute:
+```python
+In [2]: model_input_table.history
+Out [2]:
+[
+    {
+        'timestamp': 1739891304488,
+        'operation': 'WRITE',
+        'operationParameters': {'mode': 'Overwrite'},
+        'operationMetrics': {
+            'execution_time_ms': 8,
+            'num_added_files': 1,
+            'num_added_rows': 6027,
+            'num_partitions': 0,
+            'num_removed_files': 1
+        },
+        'clientVersion': 'delta-rs.0.23.1',
+        'version': 1
+    },
+    {
+        'timestamp': 1739891277424,
+        'operation': 'WRITE',
+        'operationParameters': {'mode': 'Overwrite'},
+        'clientVersion': 'delta-rs.0.23.1',
+        'operationMetrics': {
+            'execution_time_ms': 48,
+            'num_added_files': 1,
+            'num_added_rows': 6027,
+            'num_partitions': 0,
+            'num_removed_files': 0
+        },
+        'version': 0
+    }
+]
+```
+
+You can also inspect the loaded version of the table with the following method:
+
+```python
+In [3]: model_input_table.get_loaded_version()
+Out [3]: 1
+```
+
+## Using Delta tables with Spark
+
+You can also use [`PySpark`](https://spark.apache.org/docs/latest/api/python/index.html) to interact with Delta tables in your Kedro project. To set up Delta tables with Spark, consult the [documentation on the integration of Spark with Kedro](./pyspark_integration.md#spark-and-delta-lake-interaction).
+
+We recommend the following workflow, which makes use of the [transcoding feature in Kedro](../data/data_catalog_yaml_examples.md#read-the-same-file-using-different-datasets-with-transcoding):
+
+* To create a Delta table, use a `spark.SparkDataset` with `file_format="delta"`. You can also use this type of dataset to read from a Delta table or overwrite it.
+* To perform [Delta table deletes, updates, and merges](https://docs.delta.io/latest/delta-update.html#language-python), load the data using a `DeltaTableDataset` and perform the write operations within the node function.
+
+As a result, we end up with a catalog that looks like this:
+
+```yaml
+temperature:
+  type: spark.SparkDataset
+  filepath: data/01_raw/data.csv
+  file_format: "csv"
+  load_args:
+    header: True
+    inferSchema: True
+  save_args:
+    sep: '|'
+    header: True
+
+weather@spark:
+  type: spark.SparkDataset
+  filepath: s3a://my_bucket/03_primary/weather
+  file_format: "delta"
+  save_args:
+    mode: "overwrite"
+    versionAsOf: 0
+
+weather@delta:
+  type: spark.DeltaTableDataset
+  filepath: s3a://my_bucket/03_primary/weather
+```
+
+The `DeltaTableDataset` does not support `save()` operation, as the updates happen in place inside the node function, i.e. through `DeltaTable.update()`, `DeltaTable.delete()`, `DeltaTable.merge()`.
```

Comment:
Maybe

````suggestion
```{note}
The `DeltaTableDataset` does not support `save()` operation. Instead, pick the operation you want to perform (`DeltaTable.update()`, `DeltaTable.delete()`, `DeltaTable.merge()`) and write it in your node code instead.
```
````